test/cli/test_cli_functions.sh

```diff
@@ -284,10 +284,89 @@ function obc_cycle {
+    local crd_status_array=($(test_noobaa silence crd status &>/dev/stdout | awk '{print $6}' | grep -v "Exists" | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    crd_array=($(kubectl get crd | awk '{print $1}' | grep -v "NAME"))
+    
+    #comparing crd status and create arrays
+    echo_time "checking if crd noobaa create command equals to crd status command"
+    for crd_status_array in ${crd_array[@]}
+    do
+        if [[ ${crd_create_array[@]} =~ ${crd_status_array} ]]
+        then            
+            echo_time "✅ ${crd_status_array} exists in the crds list"
+        else
+            echo_time "❌ ${crd_status_array} is not existed in the crds list"         
+        fi               
+    done
+         
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi      
+
+    test_noobaa timeout crd delete 
+    local crd_status_after_delete=($(test_noobaa silence crd status &>/dev/stdout | awk '{print $6}' | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    #checking if crds still exist in the test after the delete
+    for crd in ${crd_array[@]} 
+    do
+        if [[ ${crd_status_after_delete[@]} =~ ${crd} ]]
+        then    
+            echo_time "❌ crd ${crd} still exists in the test. exiting."
+            exit 1       
+        else 
+            echo_time "✅ crd ${crd} deleted from test" 
+        fi
+    done
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi 
+
+    echo_time "creating crd again for checks..."     
+    test_noobaa timeout crd create 
+    local crd_after_create=($(test_noobaa silence crd status &>/dev/stdout | awk '{print $6}' | sed -e 's/[""\\]//g' | sed 's/.$//'))
+    #comparing crds after running create command
+    for crd_after_create in ${crd_array[@]}
+    do
+        if [[ ${crd_after_create[@]} =~ ${crd_status_array} ]]
+        then            
+            echo_time "✅ ${crd_after_create} is exsisted in the crds list"
+        else
+            echo_time "❌ ${crd_after_create} is not existed in the crds list"         
+        fi               
+    done
+
+    kuberun get crd  
+    if [ $? -ne 0 ]
+    then
+        echo_time "❌ kubectl get crd failed"
+        exit 1 
+    else 
+        echo_time "✅ kubectl get crd passed"    
+    fi 
+
+    test_noobaa timeout crd status
+       
+    echo_time "✅  crd cycle is done"
+}
+
 function delete_backingstore_path {
     local object_bucket backing_store
     local backingstore=($(test_noobaa silence backingstore list | grep -v "NAME" | awk '{print $1}'))
-    local bucketclass=($(test_noobaa silence bucketclass list  | grep ${backingstore[1]} | awk '{print $1}'))
+    local bucketclass=($(test_noobaa bucketclass list  | grep ${backingstore[1]} | awk '{print $1}'))
+    echo "${bucketclass}"
```

Comment:
Why are we printing the bucketclass here?

Reply:
I will fix it for the next pr. Tried to check something here and forgot to delete it

Reply:
Next commit 🙂